internal/signing/channelaccounts/channel_accounts_model.go

```diff
@@ -0,0 +1,150 @@
+package channelaccounts
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/wallet-backend/internal/db"
+)
+
+const ChannelAccountWaitTime = 6
+
+var (
+	ErrNoAvailableChannelAccount = errors.New("no available channel account")
+	ErrChannelAccountNotFound    = errors.New("channel account not found")
+)
+
+type ChannelAccount struct {
+	PublicKey           string       `db:"public_key"`
+	EncryptedPrivateKey string       `db:"encrypted_private_key"`
+	UpdatedAt           time.Time    `db:"updated_at"`
+	CreatedAt           time.Time    `db:"created_at"`
+	LockedAt            sql.NullTime `db:"locked_at"`
+	LockedUntil         sql.NullTime `db:"locked_until"`
+}
+
+type ChannelAccountStore interface {
+	GetIdleChannelAccount(ctx context.Context, lockedUntil time.Duration) (*ChannelAccount, error)
+	Get(ctx context.Context, sqlExec db.SQLExecuter, publicKey string) (*ChannelAccount, error)
+	BatchInsert(ctx context.Context, sqlExec db.SQLExecuter, channelAccounts []*ChannelAccount) error
+	Count(ctx context.Context) (int64, error)
+}
+
+type ChannelAccountModel struct {
+	DB db.ConnectionPool
+}
+
+var _ ChannelAccountStore = (*ChannelAccountModel)(nil)
+
+func (ca *ChannelAccountModel) GetIdleChannelAccount(ctx context.Context, lockedUntil time.Duration) (*ChannelAccount, error) {
+	query := fmt.Sprintf(`
+		UPDATE channel_accounts
+		SET 
+			locked_at = NOW(),
+			locked_until = NOW() + INTERVAL '%d seconds'
+		WHERE public_key = (
+			SELECT
+				public_key
+			FROM channel_accounts
+			WHERE 
+				locked_until IS NULL
+				OR locked_until < NOW()
+			ORDER BY random()
+			LIMIT 1
+			FOR UPDATE SKIP LOCKED
+		)
+		RETURNING *
+	`, int64(lockedUntil.Seconds()))
+
+	var channelAccount ChannelAccount
+	for range ChannelAccountWaitTime {
```

Comment:
👋 this retry logic should 100% not be in the model. You should remove it and address it in another PR if needed.